packages/synapse-react-client/src/components/useImmutableTableQuery.ts

```diff
@@ -18,18 +19,36 @@ import {
   queryRequestsHaveSameTotalResults,
   removeEmptyQueryParams,
 } from '../utils/functions/queryUtils'
+import { ReadonlyDeep } from 'type-fest'
+import useCommittedState from '../utils/hooks/useCommittedState'
+import { useDebouncedEffect } from '@react-hookz/web'
+
+type QueryChangeCommitOptions =
+  | {
+      // This and future changes including debounce will not be committed until the debounceDelay has elapsed
+      debounce: true
+    }
+  | {
+      // The change will not be committed until the commit function is invoked.
+      noCommit: true
+    }
```

Comment:
Used an object instead of a string literal in case we wanted to add more options, like a way to manually control the debounce duration.